components/files/file_list/file_list.py

```diff
@@ -1,25 +1,15 @@
 #!/usr/bin/python
-import argparse
-import json
-import logging
 import os
 import sys
-import requests
-
-sys.path.append(os.path.join(os.path.dirname(os.path.realpath(__file__)), "..", "..", "tokens"))
-sys.path.append(os.path.join(os.path.dirname(os.path.realpath(__file__)), "..", "..", "utils"))
-sys.path.append(os.path.join(os.path.dirname(os.path.realpath(__file__)), "..", "..", "utils", "parameters", "pagination"))
-sys.path.append(os.path.join(os.path.dirname(os.path.realpath(__file__)), "..", "..", "metadata", "rdm_parameters", "rdm_pagination"))
-sys.path.append(os.path.join(os.path.dirname(os.path.realpath(__file__)), "..", "..", "metadata", "metadata_list"))
-sys.path.append(os.path.join(os.path.dirname(os.path.realpath(__file__)), "..", "..", "metadata"))
-
-from get_token import *
-from utils import *
-from args import *
-from rdm_pagination_traits import *
-from metadata_traits import *
-from metadata_list  import *
-from pagination import *
+
+here = os.path.dirname(os.path.realpath(__file__))
+components_dir = here[:here.rindex("components")+11]

```

Comment:
Thinking a bit more on this, you're using `join` so you don't need to preserve the / at the end, and this code is ugly enough that it's probably worth tidying it away into a function even if we have to copy-paste it everywhere:

```python
def up2(path, last, inclusive=True, sep=os.path.sep):
    return path[:path.rindex(sep + last + sep) + (1+len(last) if inclusive else 0)]

here = os.path.dirname(os.path.realpath(__file__))
incl = up2(here, "components")         # has "components" at the end
excl = up2(here, "components", False)  # chops off "components"
```
`inclusive` is for the `workflows` cases:

```python
components_dir = os.path.join(up2(here, "workflows", False), "components")
```
n.b.  `sep+last+sep` forces the `last` to be a full component, so for example 

```python
here = "/yada/yada/components/extra-components/wibble"
print(up2(here, "components"))
```
will print 
```
/yada/yada/components
```
not
```
/yada/yada/components/extra-components
```
(`sep` is only a parameter to avoid an extra line to set it, and also to shorten the code. You shouldn't give it a value).

Reply:
(maybe the `1+` in `up2` should be `len(sep)+`. Then you could use an arbitrary separator. But in practice you never will.)

Reply:
Agree with all of these recommendations. I've updated the names but will adopt the principles throughout. Thank you. This should make it more concise, robust and flexible.

Reply:
Agree with all of these recommendations. I've updated the names but will adopt the principles throughout. Thank you. This should make it more concise, robust and flexible.